zilencer/tests/test_stripe.py

```diff
@@ -530,6 +530,55 @@ def check_subscription_save(subscription: stripe.Subscription, idempotency_key:
         do_activate_user(user)
         check_billing_processor_update(RealmAuditLog.USER_ACTIVATED, self.quantity + 1)
 
+    @mock.patch("stripe.Customer.save")
+    @mock.patch("stripe.Customer.retrieve", side_effect=mock_customer_with_subscription)
+    @mock.patch("stripe.Customer.create", side_effect=mock_create_customer)
+    @mock.patch("stripe.Subscription.create", side_effect=mock_create_subscription)
+    def test_card_change(self, mock_create_subscription: mock.Mock,
+                         mock_create_customer: mock.Mock, mock_retrieve_customer: mock.Mock,
+                         mock_save_customer: mock.Mock) -> None:
+        user = self.example_user("hamlet")
+        self.login(user.email)
+        response = self.client_get("/upgrade/")
+
+        self.client_post("/upgrade/", {
+            'stripeToken': self.token,
+            'signed_seat_count': self.get_signed_seat_count_from_response(response),
+            'salt': self.get_salt_from_response(response),
+            'plan': Plan.CLOUD_ANNUAL})
+
+        RealmAuditLog.objects.filter(acting_user=user).delete()
+
+        result = self.client_post("/json/billing/sources/change", {'stripe_token': ujson.dumps(self.token)})
+        self.assert_json_success(result)
+        mock_save_customer.assert_called_once_with()
```

Comment:
Might be good to check that we called save with the new source. You can see how it's done by grepping this file for `mock.patch.object`.